MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/CreateCapillaryPressureModel.cpp

```diff
@@ -30,56 +28,76 @@ namespace PorousMedium
                   including <type>BrookCorey</type>
                   and it has a tag of <capillary_pressure>
 */
-static std::unique_ptr<CapillaryPressureSaturation>
-                         createBrookCorey(BaseLib::ConfigTree const& config)
+static std::unique_ptr<CapillaryPressureSaturation> createBrookCorey(
+    BaseLib::ConfigTree const& config)
 {
-    std::array<double, 5> parameters = {
-        {//! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__pd}
-         config.getConfigParameter<double>("pd"),
-         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__sr}
-         config.getConfigParameter<double>("sr"),
-         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__smax}
-         config.getConfigParameter<double>("smax"),
-         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__m}
-         config.getConfigParameter<double>("m"),
-         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__pc_max}
-         config.getConfigParameter<double>("pc_max")}};
-
-    assert(parameters[3] >= 1.0);  // m >= 1
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type}
+    config.checkConfigParameter("type", "BrookCorey");
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__pd}
```

Comment:
If I'm not mistaken the `type` in the string must be `BrookCorey` because it is specific for that model.
@chleh Please correct me if I'm talking garbage.